src/precice/SolverInterface.cpp

```diff
@@ -1,26 +1,34 @@
 #include "precice/SolverInterface.hpp"
+#include <string_view>
 #include "cplscheme/Constants.hpp"
 #include "precice/impl/SolverInterfaceImpl.hpp"
 #include "precice/impl/versions.hpp"
 
 namespace precice {
 
+namespace {
+std::string_view toSV(precice::string_view sv)
+{
+  return {sv.data(), sv.size()};
+}
+} // namespace
```

Comment:
Wouldn't it be way easier to define an implicit conversion operator from `precice::span` to `std::span`? Or do you want to make the conversion explicit? Adds a bit of visual clutter IMO.

Reply:
Agreed.
I would have preferred an implicit conversion, but I couldn't find a way how to add an implicit conversion as a non-member function.

I could add it to the span class, but this would enable it everywhere and make upgrading the span more difficult/error-prone. 
Given that the conversion is only used in this one cpp file, I added a separate function.

Reply:
I ran into issues where the span allows NULL characters inside the span.
So we also need to trim the view in the conversion. This kind of justifies a separate function.

Reply:
This reason for this was also new to me. :wink:  

`span<const char>("123")` dispatches to the overload that handles arrays `template <class T, int N> span(T[N])`.
The C string literal is a C array `char const[4]` with value `{ '1', '2', '3', '\0' }`, hence size 4.

Passing the c string literal as a `const char *` however will dispatch to the c string constructor `span(const char *)`, which uses `std::strlen` to determine the size of the string.

So, we cannot dispatch directly to the `std::string_view(char*, size)` constructor, as the span may include a NULL char. Thus, we need to extract everything until the first (possibly absent) `\0` char.

Reply:
Thanks for the example, might be worth a small note in the code, unless you want anyone else in the future to unravel this.